Wip/bhahn/imx95 usb (#472)
phytec/doc-bsp-yocto: source/bsp/imx9/imx95-fpsc/head.rst

And the additions on the carrierboard:
:linux-phytec-imx:`tree/v6.18.20-2.0.0-phy/arch/arm64/boot/dts/freescale/imx95-phyflex-libra-rdk.dts#L436`

> USB-A (Host)

Reviewer: I'd prefer keeping the original header or using "USB Host Interface" or even just "USB Host", as this chapter is about the host interface, not about the USB type.

Author: How about doing only one USB chapter instead, where I have a general board section where I describe all the USB ports that are available on Libra and then two subchapters for USB Device and USB Host. I will push what I imagine here.

------------

The |kit| has two USB 2.0 type A ports (X16) that are configured as USB host

> role.

Reviewer: nitpick: Maybe "interface" is the correct wording here?
```suggestion
interface.
```

.. include:: /bsp/peripherals/usb-host.rsti

DT representation for USB-A Host Ports:
:linux-phytec-imx:`tree/v6.18.20-2.0.0-phy/arch/arm64/boot/dts/freescale/imx95-phyflex-libra-rdk.dts#L538`

> USB-C (OTG)

Reviewer: ```suggestion
USB Type-C DRD
```

Author: DRP?!

Author: Heading got removed

-----------

> The |kit| has a USB-C 3.0 OTG Port (X18) that is a dual role port and can act

Reviewer: I think it should be:
```suggestion
The |kit| has a USB Type-C port (X18) supporting USB 3.2 Gen 1 (5 Gbps) with Dual-Role Data (DRD) that can act
```
"USB 3.0" is obselete and got renamed, AFAIK. Also "OTG" seems to be old USB 2.0 nomenclature and got replaced with DRD. (The USB standard is a mess, I know.)

Proof-reading through an AI may help here, actually.

Author: Yes, USB 3.0 is "old" and got renamed to USB3.1 Gen1 and then to USB3.2 Gen1x1. I decided to keep the USB3.0, as everybody knows what USB3.0 is, but I would not have known what USB3.2 Gen1x1 is. And as far as I know the function is the same between those two standards. It is only a renaming to fit into the newer USB3.2 standard.
You mean DRP (Dual Role Port) not DRD?!

Reviewer: It's DRD (dual role device). But I'm not sure now, whether that's actually applicable here. DRD can transfer data as host and device simultaneously, whereas OTG can only act as one at a time. I think you know better what is now (probably OTG still). Let's keep it as is. :see_no_evil:

as device or host. In Linux, the Port will switch the role automatically
depending what is plugged into the port. In U-Boot, the Port will be
initialized as device or as host depending on the command that is used.

> Otherwise USB host works the same as for the two USB-A ports.

Reviewer: "USB-A" is not an official term... 😬 Instead, use "USB Standard-A" connector:
```suggestion
Otherwise USB host works the same as for the two USB Standard-A ports.
```

Author: I don't think anybody sais USB Standard-A. I can use USB Type-A instead. I think that is the more common phrasing.

Author: I removed the sentence entirely, as I don't think it is really necessary here.

Author: USB-A might not be an official phrase, I don't actually know that, but it is commonly used throughout all our BSP manuals and everyone understands what is meant by it. And we are not the only ones that use it, as it seems to be a quite common expression.
